src/java/org/apache/cassandra/service/reads/DataResolver.java

```diff
@@ -278,6 +277,10 @@ private PartitionIterator resolveWithReplicaFilteringProtection(E replicas, Repa
 
     private  UnaryOperator<PartitionIterator> preCountFilterForReplicaFilteringProtection()
     {
+        // Key columns are immutable and should never need to participate in replica filtering
+        if (!command.rowFilter().hasNonKeyExpressions())
+            return results -> results;
```

Comment:
Early port of CASSANDRA-19938...washes out next trunk rebase...

Reply:
I think you sent me this fix at one point. That should be fine right since one we rebase the history will look correct? Do we not need the fix that this does?

Reply:
Yeah, this will be fine on the next rebase either way. Might as well leave it in.